imod/tests/test_mf6/test_mf6_resampling.py

```diff
@@ -0,0 +1,163 @@
+from datetime import datetime
+
+import pandas as pd
+
+from imod.util.expand_repetitions import resample_timeseries
+
+
+def test_timeseries_resampling():
+    # In this test, we resample a timeseries for a coarser output discretization.
+    # The output times are a subset of the input times.
+    # fmt: off
+    data = [
+        {  "time": datetime(1989, 1, 1), "rate": 100, "x": 0,"y": 0, "id": "ID", "filt_top": 20, "filt_bot": 10, },  # output time 1
+        {"time": datetime(1989, 1, 3), "rate": 200, "x": 0, "y": 0,  "id": "ID", "filt_top": 20, "filt_bot": 10,},
+        { "time": datetime(1989, 1, 4), "rate": 300, "x": 0, "y": 0, "id": "ID", "filt_top": 20, "filt_bot": 10, },
+        {"time": datetime(1989, 1, 5), "rate": 400, "x": 0, "y": 0,"id": "ID", "filt_top": 20, "filt_bot": 10, },  # output time 2
+        { "time": datetime(1989, 1, 6), "rate": 500, "x": 0,  "y": 0,"id": "ID","filt_top": 20, "filt_bot": 10,}, # output time 3
+    ]  
+    # fmt: on
+    # initialize data of lists.
+
+    # setup panda table
+    timeseries = pd.DataFrame(
+        data, columns=["time", "rate", "x", "y", "id", "filt_top", "filt_bot"]
+    )
```

Comment:
This is more readable, could you try and see if this works? If so, update the other cases in this form? I think this is more readable.

```suggestion
    timeseries = pd.DataFrame(
        {}, columns=["time", "rate", "x", "y", "id", "filt_top", "filt_bot"]
    )
   
    timeseries["time"] = [datetime(1989, 1, i) for i in [1, 3, 4, 5, 6]]
    timeseries["rate"] = [i * 100 for i in range(1, 6)]
    timeseries["x"] = 0
    timeseries["y"] = 0
    timeseries["id"] = "ID"
    timeseries["filt_top"] = 20
    timeseries["filt_bot"] = 10

```